test/langtools/jdk/javadoc/tool/CheckManPageOptions.java

```diff
@@ -0,0 +1,287 @@
+        out.println("Man page options: " + manPageOptions);
+        out.println();
+
+        var toolOptions = getToolOptions();
+        out.println("ToolOptions: " + toolOptions);
+        out.println();
+
+        var docletOptions = getDocletOptions();
+        out.println("DocletOptions: " + docletOptions);
+        out.println();
+
+        var toolDocletOnly = new TreeSet<String>();
+        toolDocletOnly.addAll(toolOptions);
+        toolDocletOnly.addAll(docletOptions);
+        toolDocletOnly.removeAll(manPageOptions);
+        toolDocletOnly.removeAll(MISSING_IN_MAN_PAGE);
+        if (!toolDocletOnly.isEmpty()) {
+            error(" The following options are defined by the tool or doclet, but not defined in the man page:\n"
+                    + toSimpleList(toolDocletOnly));
+        }
+
+        var manPageOnly = new TreeSet<String>();
+        manPageOnly.addAll(manPageOptions);
+        manPageOnly.removeAll(toolOptions);
+        manPageOnly.removeAll(docletOptions);
+        if (!manPageOnly.isEmpty()) {
+            error("The following options are defined in the man page, but not defined by the tool or doclet:\n"
+                    + toSimpleList(manPageOnly));
+        }
+
+        if (!MISSING_IN_MAN_PAGE.isEmpty()) {
+            var notMissing = new TreeSet<>(MISSING_IN_MAN_PAGE);
+            notMissing.retainAll(manPageOptions);
+            if (!notMissing.isEmpty()) {
+                error("The following options were declared as missing, but were found on the man page:\n"
+                        + toSimpleList(notMissing));
+            }
+
+            out.println("NOTE: the following options are currently excluded and need to be documented in the man page:");
+            out.println(toSimpleList(MISSING_IN_MAN_PAGE));
+        }
+
+        if (errors > 0) {
+            out.println(errors + " errors found");
+            throw new Exception(errors + " errors found");
+        }
+    }
+
+    int errors = 0;
+    void error(String message) {
+        ("Error: " + message).lines().forEach(out::println);
+        errors++;
+    }
+
+    String toSimpleList(Collection<String> items) {
+        return items.stream().collect(Collectors.joining(", ", "    ", ""));
+    }
+
+    Path findDefaultFile() {
+        return findRootDir().resolve("src/jdk.javadoc/share/man/javadoc.1");
+    }
+
+    Path findRootDir() {
+        Path dir = Path.of(System.getProperty("test.src", ".")).toAbsolutePath();
+        while (dir != null) {
+            if (Files.exists(dir.resolve("src"))) {
+                return dir;
+            } else {
+                Path openDir = dir.resolve("open");
+                if (Files.exists(openDir.resolve("src"))) {
+                    return openDir;
+                }
+            }
+            dir = dir.getParent();
+        }
+        throw new IllegalStateException("cannot find root dir");
+    }
+
+    List<String> getToolOptions() throws Error {
```

Comment:
I wonder if a better approach to get options would be to parse output of `javadoc -help`. After all, that is what we are after: ensure 1-1 correspondence between two documentation sources.